src/hivpy/demographics.py

```diff
@@ -113,23 +113,24 @@ def __init__(self, min_age, max_age, modelParams):
         self.min_age = min_age
         model_age_limit = -modelParams[1]/modelParams[0]
         if(max_age > model_age_limit):
-            warnings.warn("Max age exceeds the maximum age limit for this model (negative probability).\
-                            Adjusting max age to " + model_age_limit)
+            logging.getLogger("Demographics").warning(f"Max age exceeds the maximum age limit for\
+                                                       this model (negative probability).\
+                                                       Adjusting max age to {model_age_limit}")
```

Comment:
Very minor - I think there's a space missing here between the lines?

If you want to avoid the continuation `\`, you can also break up string literals over multiple lines and they will be joined automatically:
```suggestion
            logging.getLogger("Demographics").warning(f"Max age exceeds the maximum age limit for "
                                                       f"this model (negative probability). "
                                                       f"Adjusting max age to {model_age_limit}")
```
(I may well have messed up the indentation here)